recipe/bld.bat

```diff
@@ -1,16 +1,23 @@
-mkdir "%SRC_DIR%"\build
-pushd "%SRC_DIR%"\build
+@echo on
 
-if NOT "%cuda_compiler_version%"=="None" (
+mkdir %SRC_DIR%\build
+pushd %SRC_DIR%\build
+
+if not "%cuda_compiler_version%" == "None" (
     set "CMAKE_ARGS=-DUSE_CUDA=ON %CMAKE_ARGS%"
+
+    :: Workaround CUB build errors on CUDA 11.8
+    if "%cuda_compiler_version%" == "11.8" (
+        set "CUDAFLAGS=%CUDAFLAGS% -DWIN32_LEAN_AND_MEAN"
+    )
```

Comment:
So this helps the Windows CUDA 11.8 build progress

Though now see a new error

```
%SRC_DIR%\src\data\adapter.h(289): here



Internal error: assertion failed at: "overload.c", line 28946 in deduce_class_template_args





1 catastrophic error detected in the compilation of "D:/bld/xgboost-split_1721471361683/work/src/data/ellpack_page_source.cu".

Compilation aborted.

ellpack_page_source.cu

nvcc error   : 'cudafe++' died with status 0xC0000409 
```